### DISCUSSION TITLE
Ability to install globally

### DISCUSSION THREAD

**Discussion by victorlap** [stefanzweifel/laravel-stats#197] [Ideas]
Hi!

It would be really cool to install this globally. That would let you discover the stats for some projects where you might not want to add this dependency. Do you think this would be something this project could support?

**Comment by stefanzweifel** [comment]
Hey 👋 

Unfortunately this is currently not possible, as `stats` is an Artisan Command. Artisan Commands have to be registered inside a Laravel application. (Could be that I'm wrong here. Feel free to prove me wrong :))

To make this work the *stats* would have to be converted into framework-agnostic CLI tool. I've written down some thoughts on this in #153.

I'm currently quite occupied with other things and didn't have the time yet to work on this. For now you have to add `stats` as a dependency.

Is adding the package as a dependeny currently a thing that prevents you from using *stats*? If so, could you give some information why that is?

(I personally are also not the biggest fan of globally installed packages. It ususally only leads to dependency conflicts between all globally installed dependencies)

**Reply by victorlap** [reply]
Hey! I think making the project framework agnostic can be a step even after that you seperate it to a global requirement / seperate binary. 

The only interaction with the application you have at the moment is a way to register the command and to publish the config. 

The reason I'm asking is because I work on a lot of applications at the same time (digital agency kind of thing) and installing the dependency in each project would also mean that we would have to keep it up-to-date in all projects. 

Maybe a global installed package would also not be ideal, but something based on laravel-zero could work. Maybe even distributed as a phar. (Other problems then arise, mainly in distributing the binary, but nothing that cannot be solved).

**Reply by stefanzweifel** [reply]
Thanks for the explanation. Might give Laravel Zero a shot here.
